src/ralph/attachments/views.py

```diff
@@ -84,3 +75,24 @@ def formset_invalid(self, formset):
         return self.render_to_response(
             self.get_context_data(formset=formset)
         )
+
+
+class ServeAttachment(View):
+
+    def get(self, *args, **kwargs):
+        """
+        All attachments are serving by this view because we need full
+        control (e.g., permissions, rename).
+        """
+        # TODO: respect permissions
+        obj = get_object_or_404(
+            Attachment,
+            id=self.kwargs['id'],
```

Comment:
why not kwargs['id']? you could use another definition too imo: `def get(self, request, id, filename)`